requirements-dev.txt

```diff
@@ -1,7 +1,7 @@
 # The Ubuntu GUI exits immediately without error in 6.11.0.
```

Comment:
Arbitrary line to allow dependabot discussion

Reply:
> As a followup, I suggest me adding a Github workflow that uses [Dependabot](https://github.com/dependabot) to automatically bump dependencies as they become available on Pypi.

> Edit: I see you have used dependabot in the past. What was the reason for removing it?

I believe we removed it because we didn't have an appropriate level of automated test coverage to verify dependency bumps.

Reply:
That makes sense, I'll refrain from adding that workflow.

Reply:
With the addition of the conformance suites we can probably begin using dependabot along with a generous use of the ignore list to filter out anything we think likely requires manual testing (especially anything GUI related). 

As an aside, Arelle is a little bit of a strange beast in that it's both an application which should have pinned dependencies (this PR is a good first step) and is used as a library which should also separately have dependency version ranges (we'll need to set this up before publishing the official pypi package).

Ideally we'd use pipenv or poetry to lock transitive dependencies for the application/local dev side of things. @ggravlingen as an outside contributor would the use of pipenv or poetry introduce any potential friction for you using or contributing to the project?

Reply:
My current use case is to run `pip install git+https://github.com/Arelle/Arelle.git#0.9.91` along with pandas and lxml. I'm then able to use the library to parse XBRL-files in the ESEF format.

I have never used pipenv or poetry but just having had a quick look, I don't believe using those tools would affect how I'm using this library.